### ISSUE TITLE
Godot crash on startup under linux

### ISSUE THREAD

**Issue by thennothinghappened** [godotengine/godot#70177]
### Godot version

4.0.beta8.official.45cac42c0

### System information

Arch Linux, NVIDIA GeForce RTX 3070 (525.60.11-3), x11/OpenGL

### Issue description

Booting any version of Godot, the project list window appears for a few seconds, then crashes.
```
$ godot4
Godot Engine v4.0.beta8.official.45cac42c0 - https://godotengine.org
OpenGL Renderer: NVIDIA GeForce RTX 3070/PCIe/SSE2
 
[xcb] Unknown sequence number while processing queue
[xcb] You called XInitThreads, this is not your fault
[xcb] Aborting, sorry about that.
godot4: xcb_io.c:278: poll_for_event: Assertion `!xcb_xlib_threads_sequence_lost' failed.
zsh: IOT instruction (core dumped)  godot4
```

I was originally trying to use the Godot 4 Beta, however the issue seems to happen on stable (3.5.1.stable.arch_linux) as well.

This is definitely an issue on my system that's appeared recently, as I've tested the alpha versions of v4 previously without issue. I'm not sure what exactly has gone wrong though.

### Steps to reproduce

1) Start Godot

### Minimal reproduction project

N/A

**Comment by thennothinghappened** [comment]
Here's a log from stable with `--verbose`:
```
$ godot --verbose
Godot Engine v3.5.1.stable.arch_linux - https://godotengine.org
XInput: Refreshing devices.
XInput: No touch devices found.
Detecting GPUs, set DRI_PRIME in the environment to override GPU detection logic.
Only one GPU found, using default.
Using GLES3 video driver
OpenGL ES 3.0 Renderer: NVIDIA GeForce RTX 3070/PCIe/SSE2
Async. shader compilation: OFF
OpenGL ES 2D Batching: ON
Batching Options:
	max_join_item_commands 16
	colored_vertex_format_threshold 0.25
	batch_buffer_size 16384
	light_scissor_area_threshold 1
	item_reordering_lookahead 4
	light_max_join_items 32
	single_rect_fallback False
	debug_flash False
	diagnose_frame False
PulseAudio: context other
PulseAudio: context other
PulseAudio: context other
PulseAudio: context ready
PulseAudio: Detecting channels for device: alsa_output.pci-0000_0b_00.1.hdmi-stereo-extra1
PulseAudio: detected 2 output channels
PulseAudio: audio buffer frames: 512 calculated output latency: 11ms
JoypadLinux: udev enabled and loaded successfully.
 
CORE API HASH: 6609155589855387091
EDITOR API HASH: 17280644357449250417
Loading resource: res://.config/godot/editor_settings-3.tres
EditorSettings: Load OK!
[xcb] Too much data requested from _XRead
[xcb] This is most likely caused by a broken X extension library
[xcb] Aborting, sorry about that.
godot: xcb_io.c:833: _XRead: Assertion `!xcb_xlib_too_much_data_requested' failed.
zsh: IOT instruction (core dumped)  godot --verbose
```

**Comment by thennothinghappened** [comment]
And again, a log from v4 with `--verbose`:
```
godot4 --verbose
Godot Engine v4.0.beta8.official.45cac42c0 - https://godotengine.org
TextServer: Added interface "Dummy"
TextServer: Added interface "ICU / HarfBuzz / Graphite (Built-in)"
XInput: Refreshing devices.
XInput: No touch devices found.
Detecting GPUs, set DRI_PRIME in the environment to override GPU detection logic.
Only one GPU found, using default.
Using "default" pen tablet driver...
OpenGL Renderer: NVIDIA GeForce RTX 3070/PCIe/SSE2
JoypadLinux: udev enabled and loaded successfully.
PulseAudio: context other
PulseAudio: context other
PulseAudio: context other
PulseAudio: context ready
PulseAudio: Detecting channels for device: alsa_output.pci-0000_0b_00.1.hdmi-stereo-extra1
PulseAudio: detected 2 output channels
PulseAudio: audio buffer frames: 512 calculated output latency: 11ms
 
TextServer: Primary interface set to: "ICU / HarfBuzz / Graphite (Built-in)".
CORE API HASH: 3230741874
EDITOR API HASH: 959301086
EditorSettings: Load OK!
Loaded builtin certs
EditorSettings: Save OK!
[xcb] Unknown sequence number while processing queue
[xcb] You called XInitThreads, this is not your fault
[xcb] Aborting, sorry about that.
godot4: xcb_io.c:278: poll_for_event: Assertion `!xcb_xlib_threads_sequence_lost' failed.
zsh: IOT instruction (core dumped)  godot4 --verbose
```

**Comment by thennothinghappened** [comment]
I've had a go trying different versions of libX11 to see if that'd fix it, rolling back to `1.7.0-1` fixed the problem, so I guess this is just an issue with libX11.

**Comment by akien-mga** [comment]
For the record, duplicate of https://github.com/godotengine/godot/issues/69352
